src/Main.elm

```diff
@@ -1,10 +1,405 @@
 module Main exposing (main)
 
+import Array
+import Browser
+import Browser.Dom
 import Data.Audience
 import Data.AudienceFolder
-import Html exposing (Html)
+import Html
+import Html.Attributes
+import Html.Events
+import Lazy.Tree
+import Lazy.Tree.Zipper
+import Task
 
 
-main : Html msg
+
+-- MAIN
+
+
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.element
+        { init = \_ -> init
+        , update = update
+        , view = view
+        , subscriptions = \_ -> Sub.none
+        }
+
+
+
+-- MODEL
+
+
+type Node
+    = Audience Data.Audience.Audience
+    | AudienceFolder Data.AudienceFolder.AudienceFolder
+
+
+type alias Model =
+    -- An item of the `zipper` stores whether it is open, whether it is selected, and the audience/folder properties.
+    { zipper : Lazy.Tree.Zipper.Zipper ( Bool, Bool, Node )
```

Comment:
I'd prefer either some custom types to give the bools a name (mumble mumble _boolean identity crisis_ mumble mumble), or at changing this 3-tuple into a record. I know "what is what" is described in the comments but records seem better to me in that regard.

Reply:
I agree with @Janiczek 's comment.